stubs/jsonschema/jsonschema/_format.pyi

```diff
@@ -1,35 +1,42 @@
-from typing import Any
+from typing import Any, Iterable
 
 class FormatChecker:
     checkers: Any
-    def __init__(self, formats: Any | None = ...) -> None: ...
+    def __init__(self, formats: Iterable[str] | None = ...) -> None: ...
     def checks(self, format, raises=...): ...
     cls_checks: Any
     def check(self, instance, format) -> None: ...
-    def conforms(self, instance, format): ...
+    def conforms(self, instance, format) -> bool: ...
 
-draft3_format_checker: Any
-draft4_format_checker: Any
-draft6_format_checker: Any
-draft7_format_checker: Any
+draft3_format_checker: FormatChecker
+draft4_format_checker: FormatChecker
+draft6_format_checker: FormatChecker
+draft7_format_checker: FormatChecker
+draft201909_format_checker: FormatChecker
+draft202012_format_checker: FormatChecker
 
-def is_email(instance): ...
-def is_ipv4(instance): ...
-def is_ipv6(instance): ...
-def is_host_name(instance): ...
-def is_idn_host_name(instance): ...
-def is_uri(instance): ...
-def is_uri_reference(instance): ...
-def is_iri(instance): ...
-def is_iri_reference(instance): ...
-def is_datetime(instance): ...
-def is_time(instance): ...
-def is_regex(instance): ...
-def is_date(instance): ...
-def is_draft3_time(instance): ...
-def is_css_color_code(instance): ...
-def is_css21_color(instance): ...
-def is_css3_color(instance): ...
-def is_json_pointer(instance): ...
-def is_relative_json_pointer(instance): ...
-def is_uri_template(instance, template_validator=...): ...
+def is_email(instance) -> bool: ...
+def is_ipv4(instance) -> bool: ...
+def is_ipv6(instance) -> bool: ...
+
+# is_host_name is only defined if fqdn is installed.
```

Comment:
Nit: It would be better to put these comments to the allowlist. You can add comments to end of line, e.g.

```
jsonschema._format.is_host_name  # defined only if fqdn is installed
```

Reply:
I actually like them in the source file as well as the allowlist is basically invisible when developing with the stubs, while the comment in the source are visible when jumping into the stub file.